lib/asm/wsman.rb

```
@@ -1073,6 +1075,11 @@ def power_off
 
       if power_state != :off
         set_power_state(:requested_state => :off)
+        (1..30).each do |wait_counter|
+          break if power_state == :off
+          logger.debug "Server is not in power-off state. Retry counter %d" % wait_counter if logger
+          sleep 10
+        end
```

I am still wondering what happens at the end of the retry loop if the machine is still on? 

To me this should raise an error - and the docs set that expectation by stating a exception is raised when the command fails.

Would that make sense? As it is now the machine can still be on and no-one would be told, hard to say whats best since that will be a behaviour change so we could possibly do that in another ticket, @gavin-scott any ideas?